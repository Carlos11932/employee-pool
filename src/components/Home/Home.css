
.home-container {
    max-width: 800px;
    margin: 20px auto;
    padding: 20px;
    background: white;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    text-align: center;
  }
  
  .home-container h3 {
    margin-bottom: 10px;
  }
  
  .home-container ul {
    list-style: none;
    padding: 0;
  }
  
  .home-container ul li {
    margin-bottom: 10px;
    padding: 10px;
    background: #f9f9f9;
    border: 1px solid #ddd;
    border-radius: 4px;
  }
  
  .home-container ul li a {
    text-decoration: none;
    color: #007bff;
  }
  
  .home-container ul li a:hover {
    text-decoration: underline;
  }
  
  .toggle-buttons {
    margin-bottom: 20px;
  }
  
  .toggle-buttons button {
    margin: 0 10px;
    padding: 10px 20px;
    cursor: pointer;
  }
  
  .questions-list {
    margin-top: 20px;
  }
  
  .question-item {
    margin-bottom: 15px;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 5px;
  }