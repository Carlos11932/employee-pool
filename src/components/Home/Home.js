import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import './Home.css';

const Home = () => {
  const [showAnswered, setShowAnswered] = useState(false);
  const authedUser = useSelector(state => state.authedUser);
  const users = useSelector(state => state.users);
  const questions = useSelector(state => state.questions);

  if (!authedUser || !users[authedUser] || !questions) {
    return <div>Loading...</div>;
  }

  const answeredQuestions = Object.keys(users[authedUser].answers)
    .sort((a, b) => questions[b]?.timestamp - questions[a]?.timestamp);
  const unansweredQuestions = Object.keys(questions)
    .filter(qid => !answeredQuestions.includes(qid))
    .sort((a, b) => questions[b]?.timestamp - questions[a]?.timestamp);

  return (
    <div className="home-container">
      <div className="toggle-buttons">
        <button onClick={() => setShowAnswered(false)} disabled={!showAnswered}>Unanswered Questions</button>
        <button onClick={() => setShowAnswered(true)} disabled={showAnswered}>Answered Questions</button>
      </div>
      <div className="questions-list">
        {showAnswered ? (
          <>
            <h3>Answered Questions</h3>
            <ul>
              {answeredQuestions.map(qid => (
                questions[qid] && (
                  <li key={qid} className="question-item">
                    <Link to={`/questions/${qid}`}>{questions[qid].optionOne.text} ---------- {questions[qid].optionTwo.text}</Link>
                  </li>
                )
              ))}
            </ul>
          </>
        ) : (
          <>
            <h3>Unanswered Questions</h3>
            <ul>
              {unansweredQuestions.map(qid => (
                questions[qid] && (
                  <li key={qid} className="question-item">
                    <Link to={`/questions/${qid}`}>{questions[qid].optionOne.text} ---------- {questions[qid].optionTwo.text}</Link>
                  </li>
                )
              ))}
            </ul>
          </>
        )}
      </div>
    </div>
  );
};

export default Home;