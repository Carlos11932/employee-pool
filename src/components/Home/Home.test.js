import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Home from './Home';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Home', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      authedUser: 'sarahedo',
      users: {
        sarahedo: {
          id: 'sarahedo',
          name: 'Sarah Edo',
          avatarURL: 'Sarah_Edo.webp',
          answers: {
            "8xf0y6ziyjabvozdd253nd": 'optionOne',
            "6ni6ok3ym7mf1p33lnez": 'optionOne',
          },
          questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9'],
        },
      },
      questions: {
        "8xf0y6ziyjabvozdd253nd": {
          id: '8xf0y6ziyjabvozdd253nd',
          author: 'sarahedo',
          timestamp: 1467166872634,
          optionOne: {
            votes: ['sarahedo'],
            text: 'Build our new application with Javascript',
          },
          optionTwo: {
            votes: [],
            text: 'Build our new application with Typescript'
          }
        },
        "6ni6ok3ym7mf1p33lnez": {
          id: '6ni6ok3ym7mf1p33lnez',
          author: 'mtsamis',
          timestamp: 1468479767190,
          optionOne: {
            votes: [],
            text: 'hire more frontend developers',
          },
          optionTwo: {
            votes: ['mtsamis', 'sarahedo'],
            text: 'hire more backend developers'
          }
        }
      },
    });
  });

  it('should render unanswered questions by default', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();

    const unansweredQuestions = await screen.findAllByText('Unanswered Questions');
    expect(unansweredQuestions).not.toHaveLength(0);
  });

  it('should switch to answered questions when clicked', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(await screen.findByText('Answered Questions'));
    const answeredQuestions = await screen.findAllByText('Answered Questions');
    expect(answeredQuestions).not.toHaveLength(0);
  });
});