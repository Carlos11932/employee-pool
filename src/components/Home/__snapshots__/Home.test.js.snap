// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render unanswered questions by default 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <div
      class="toggle-buttons"
    >
      <button
        disabled=""
      >
        Unanswered Questions
      </button>
      <button>
        Answered Questions
      </button>
    </div>
    <div
      class="questions-list"
    >
      <h3>
        Unanswered Questions
      </h3>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home should switch to answered questions when clicked 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <div
      class="toggle-buttons"
    >
      <button
        disabled=""
      >
        Unanswered Questions
      </button>
      <button>
        Answered Questions
      </button>
    </div>
    <div
      class="questions-list"
    >
      <h3>
        Unanswered Questions
      </h3>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
